lib/sidekiq/fetch.rb

```diff
@@ -44,7 +44,7 @@ def retrieve_work
         return nil
       end
 
-      queue, job = redis { |conn| conn.blocking_call(false, "brpop", *qs, TIMEOUT) }
+      queue, job = redis { |conn| conn.blocking_call(TIMEOUT + 1, "brpop", *qs, TIMEOUT) }
```

Comment:
might this be the source for `RedisClient::ReadTimeoutError` we see?  Seems new and correlated with this release. 


<Details>

```json
{
  "body": {
    "trace": {
      "frames": [
        {
          "method": "block in safe_thread",
          "lineno": 19,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/component.rb"
        },
        {
          "method": "watchdog",
          "lineno": 10,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/component.rb"
        },
        {
          "method": "run",
          "lineno": 72,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/processor.rb"
        },
        {
          "method": "process_one",
          "lineno": 81,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/processor.rb"
        },
        {
          "method": "fetch",
          "lineno": 99,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/processor.rb"
        },
        {
          "method": "get_one",
          "lineno": 87,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/processor.rb"
        },
        {
          "method": "retrieve_work",
          "lineno": 299,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.8/lib/sidekiq/pro/super_fetch.rb"
        },
        {
          "method": "with",
          "lineno": 61,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "handle_interrupt",
          "lineno": 61,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "block in with",
          "lineno": 64,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "handle_interrupt",
          "lineno": 64,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "block (2 levels) in with",
          "lineno": 65,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "block in retrieve_work",
          "lineno": 300,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.8/lib/sidekiq/pro/super_fetch.rb"
        },
        {
          "method": "get_job",
          "lineno": 312,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.8/lib/sidekiq/pro/super_fetch.rb"
        },
        {
          "method": "call",
          "lineno": 312,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.8/lib/sidekiq/pro/super_fetch.rb"
        },
        {
          "method": "strict",
          "lineno": 341,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.8/lib/sidekiq/pro/super_fetch.rb"
        },
        {
          "method": "redis",
          "lineno": 28,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/component.rb"
        },
        {
          "method": "redis",
          "lineno": 99,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/capsule.rb"
        },
        {
          "method": "with",
          "lineno": 61,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "handle_interrupt",
          "lineno": 61,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "block in with",
          "lineno": 64,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "handle_interrupt",
          "lineno": 64,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "block (2 levels) in with",
          "lineno": 65,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb"
        },
        {
          "method": "block in redis",
          "lineno": 102,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.7/lib/sidekiq/capsule.rb"
        },
        {
          "method": "block in strict",
          "lineno": 355,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.8/lib/sidekiq/pro/super_fetch.rb"
        },
        {
          "method": "blocking_call",
          "lineno": 26,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/decorator.rb"
        },
        {
          "method": "blocking_call",
          "lineno": 270,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client.rb"
        },
        {
          "method": "ensure_connected",
          "lineno": 626,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client.rb"
        },
        {
          "method": "block in blocking_call",
          "lineno": 271,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client.rb"
        },
        {
          "method": "call",
          "lineno": 16,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/middlewares.rb"
        },
        {
          "method": "block (2 levels) in blocking_call",
          "lineno": 272,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client.rb"
        },
        {
          "method": "call",
          "lineno": 31,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/connection_mixin.rb"
        },
        {
          "method": "read",
          "lineno": 96,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection.rb"
        },
        {
          "method": "with_timeout",
          "lineno": 48,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/buffered_io.rb"
        },
        {
          "method": "block in read",
          "lineno": 96,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection.rb"
        },
        {
          "method": "load",
          "lineno": 50,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/resp3.rb"
        },
        {
          "method": "parse",
          "lineno": 113,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/resp3.rb"
        },
        {
          "method": "getbyte",
          "lineno": 85,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/buffered_io.rb"
        },
        {
          "method": "ensure_remaining",
          "lineno": 114,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/buffered_io.rb"
        },
        {
          "method": "fill_buffer",
          "lineno": 122,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/buffered_io.rb"
        },
        {
          "method": "loop",
          "lineno": 122,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/buffered_io.rb"
        },
        {
          "method": "block in fill_buffer",
          "lineno": 140,
          "filename": "/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.14.0/lib/redis_client/ruby_connection/buffered_io.rb"
        }
      ],
      "exception": {
        "message": "RedisClient::ReadTimeoutError",
        "class": "RedisClient::ReadTimeoutError"
      }
    }
  },
  "environment": "production",
  "framework": "Sidekiq: 7.0.7",
  "framework": "Rails: 7.0.4.3",
}
```

</Details>

Reply:
Are you using super fetch?

Reply:
yes, but I believe an equivalent change was made there?

Reply:
Not the exact same, it’s been tweaked on main to be like this. Release on Monday.